api/v1alpha2/source_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// BundleDeploymentSource configures the sources
+// from where the bundle content needs to be
+// downloaded, and the location to unpack it.
+type BundleDeplopymentSource struct {
+	// Kind of source being passed
+
+	// +kubebuilder:validation:type=string
+	Kind SourceKind `json:"kind"`
```

Comment:
Could we use something other than "kind", e.g. sort, family, type, class? In my mind kind is for a Kubernetes resource kind.

Reply:
Since we now have an RFC, let's keep the discussion there, please. Thanks!